librz/bin/format/elf/elf_info.c

```diff
@@ -845,6 +845,75 @@ static char *get_cpu_mips(ELFOBJ *bin) {
 	return strdup(" Unknown mips ISA");
 }
 
+static char *read_arm_build_attributes(char *ptr, ut32 bytes_to_read, bool isbig) {
+	while (bytes_to_read > 0) {
+		/* TODO: Search for TAG_CPU_ARCH and Tag_CPU_ARCH_PROFILE */
+	}
+	return strdup(" Unknown arm ISA");
+}
+
+static char *read_arm_aeabi_section(char *ptr, ut32 bytes_to_read, bool isbig) {
+	while (bytes_to_read > 0) {
+		ut8 sub_subsection = *ptr;
+		ut32 sub_subsection_size = rz_read_ble32(ptr + 1, isbig);
+
+		if (sub_subsection == ARM_TAG_FILE) {
+			char *str = read_arm_build_attributes(ptr + 5, sub_subsection_size - 5, isbig);
+			if (strcmp(str, " Unknown arm ISA")) {
+				return str;
+			}
+		}
+		ptr += sub_subsection_size;
+		bytes_to_read -= sub_subsection_size;
+	}
+	return strdup(" Unknown arm ISA");
+}
+
+/**
+ * Processor-specific section types explained in:
+ * 1. https://github.com/ARM-software/abi-aa/blob/main/aaelf32/aaelf32.rst
+ * 2. https://github.com/ARM-software/abi-aa/blob/main/addenda32/addenda32.rst
+ */
+static char *get_cpu_arm(ELFOBJ *bin) {
+	RzBinElfSection *section = Elf_(rz_bin_elf_get_section_with_name)(bin, ".ARM.attributes");
+	if (!section) {
+		return strdup(" Unknown arm ISA");
+	}
+
+	ut64 offset = section->offset;
+	ut64 size = section->size;
+	if (size < 1) {
+		return strdup(" Unknown arm ISA");
+	}
+
+	char *result = malloc(size + 1);
+	if (!result) {
+		return strdup(" Unknown arm ISA");
+	}
+
+	if (rz_buf_read_at(bin->b, offset, (ut8 *)result, size) < 1) {
+		free(result);
+		return strdup(" Unknown arm ISA");
+	}
+
+	char *subsection_ptr = result + 1; // Point to the first subsection (first byte is the format-version)
+	bool isbig = Elf_(rz_bin_elf_is_big_endian)(bin);
+
+	while (subsection_ptr - result < size) {
+		char *ptr = subsection_ptr;
+		ut32 subsection_size = rz_read_ble32(ptr, isbig);
+		ptr += 4;
+
+		if (!strcmp(ptr, "aeabi")) {
+			ptr += strlen("aeabi") + 1; // +1 for the null byte
+			return read_arm_aeabi_section(ptr, subsection_size - 10, isbig);
+		}
+		subsection_ptr += subsection_size;
+	}
+
```

Comment:
Minor nitpick: Put a goto label here, and jump here in all the error cases rather than returning the same string at multiple places.